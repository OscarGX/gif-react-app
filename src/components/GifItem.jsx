export const GifItem = ({ gif  }) => {
    return (
        <div className="card">
            <img src={gif.url} alt={gif.title}></img>
            <p>{ gif.title }</p>
        </div>
    )
};