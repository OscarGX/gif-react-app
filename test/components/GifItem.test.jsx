import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Probando GifItems Components', () => {
    const gif = { title: 'Title', url: 'https://example.com/' };
    test('Debe coincidir con el snapshot', () => {
      const  { container } = render(<GifItem gif={gif}/>);
      expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen y el alt indicado', () => {
      render(<GifItem gif={gif}/>);
      const { src, alt } = screen.getByRole('img');
      expect(src).toBe(gif.url);
      expect(alt).toBe(gif.title);
    });

    test('debe mostrar el tÃ­tulo en el componente', () => {
      render(<GifItem gif={gif}/>);
      expect(screen.getByText(gif.title)).toBeTruthy();
    });
    
    
});
