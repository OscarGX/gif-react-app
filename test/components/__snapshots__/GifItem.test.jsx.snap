// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Probando GifItems Components Debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title"
      src="https://example.com/"
    />
    <p>
      Title
    </p>
  </div>
</div>
`;
